packages/server/src/login/login.controller.ts

```diff
@@ -147,12 +139,24 @@ export class LoginController {
     @Res() res: Response,
     @Query('userId') userId: number,
   ): Promise<void> {
-    const token = await this.jwtService.signAsync({ userId });
+    this.enter(res, userId);
+  }
+
+  // Set cookie and redirect to proper page
+  private async enter(res: Response, userId: number) {
+    const userCourse = await UserCourseModel.findOne({
+      where: { userId },
+    });
```

Comment:
We might want to check that the current semester of the course is the current semester so it the query doesn't return an old course. But for now it should work since it's everyone's first semester

Reply:
good point, we'll have to go through the whole thing probably to support semesters later